addons/website_livechat/models/website_visitor.py

```diff
@@ -57,8 +57,13 @@ def action_send_chat_request(self):
             operator = self.env.user
             country = visitor.country_id
             visitor_name = "%s (%s)" % (visitor.display_name, country.name) if country else visitor.display_name
+            channel_partner_to_add = [(4, operator.partner_id.id)]
+            if visitor.partner_id:
+                channel_partner_to_add.append((4, visitor.partner_id.id))
+            else:
+                channel_partner_to_add.append((4, self.env.ref('base.public_partner').id))
```

Comment:
`_link_to_partner` should update this relation to remove the public partner and add the new partner